modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/adapter/SW360ReleaseClientAdapter.java

```diff
@@ -36,9 +38,21 @@ public SW360ReleaseClientAdapter(String restUrl, boolean proxyUse, String proxyH
     public SW360Release addRelease(Artifact artifact, SW360Component sw360Component, Set<String> sw360LicenseIds, HttpHeaders header) throws AntennaException {
         SW360Release release = new SW360Release();
         SW360ReleaseAdapterUtils.prepareRelease(release, sw360Component, sw360LicenseIds, artifact);
-        return releaseClient.createRelease(release, header);
-    }
 
+        if (! SW360ReleaseAdapterUtils.isValidRelease(release)) {
+            throw new AntennaException("Can not write invalid release for " + artifact.toString());
+        }
+
+
+        final SW360Release release1 = releaseClient.createRelease(release, header);
+
+        final Optional<Path> sourceFile = artifact.askForGet(ArtifactSourceFile.class);
+        if (sourceFile.isPresent()) {
```

Comment:
shouldn't there also be a check for is `uploadSources` set to `true`?

Reply:
Yes, sorry. That was stupid :(